### ISSUE TITLE
[Self-managed stack]: Agent shows up unhealthy after upgrading from [v7.12.0 Snapshot to 8.0 Snapshot]. 

### ISSUE THREAD

**Issue by amolnater-qasource** [elastic/beats#24533]
**Kibana version:** Kibana: 8.0 Snapshot Kibana self-managed stack environment

**Host OS and Browser version:** Windows 10, All

**Build Details:**
```
    Artifact link used: https://snapshots.elastic.co/7.12.0-c84ea0e3/downloads/beats/elastic-agent/elastic-agent-7.12.0-SNAPSHOT-windows-x86_64.zip
    Build: 41196
    Commit: 8d15c288999d0cc68dfba84abd4d0d2967f78561
```

**Preconditions:**
1. 8.0 Snapshot self-managed stack environment should be available.
2. 7.12.0 Snapshot agent must be installed with default policy having System and Endpoint Security integrations.

**Steps to reproduce:**
1. Login to Kibana environment.
2. Navigate to Fleet>Agents tab.
3. Click on "Upgrade agent" under Actions.

**Actual Result:**
- Agent status after upgrade is "Unhealthy"
- "elastic-endpoint" version remains 7.12.0 Snapshot.
![10](https://user-images.githubusercontent.com/77374876/111112543-569a5980-8586-11eb-958a-fc86a498d200.PNG)

- No new data for "System" under Data Streams tab.

**Expected Result:**
Agent should be upgraded successfully and data streams should be generated for System and Endpoint after upgrade.

**Screenshots:**
![Upgrade](https://user-images.githubusercontent.com/77374876/111113244-7a11d400-8587-11eb-955d-0b34646eaa22.png)

**Logs:**
[elastic-agent-json.log](https://github.com/elastic/beats/files/6139234/elastic-agent-json.log)

**Comment by amolnater-qasource** [comment]
@manishgupta-qasource Please review.

**Comment by manishgupta-qasource** [comment]
Reviewed & Assigend to @EricDavisX

**Comment by dikshachauhan-qasource** [comment]
Hi Eric

We have performed same scenario on 7.13 snapshot Could build and found it not reproducible.

Observations:
- Agent upgraded successfully from Kibana UI from v7.12BC4 to 7.13 snapshot.
- All binaries are in running state [filebeat, metricbeat and endpoint]
-  Data streams available after upgrade.
- Host is visible under Security>admin tab.

Screenshot:
![image](https://user-images.githubusercontent.com/72375798/111145839-bc9ad700-85ae-11eb-8175-463df31d0eca.png)


Logs:
[endpoint-000000.log](https://github.com/elastic/beats/files/6140979/endpoint-000000.log)
[elastic-agent-json.log](https://github.com/elastic/beats/files/6140982/elastic-agent-json.log)

**Build details:**
```
BUILD 39526
COMMIT 6d23a9826606d065c5cbb467bb7bbe68f12da37b
Artifact link used: https://staging.elastic.co/7.12.0-336ff10d/summary-7.12.0.html 
```

Please let us know if any further information is required.

Thanks
QAS

**Comment by EricDavisX** [comment]
this is a duplicate to the other known issue:
https://github.com/elastic/beats/issues/24477

it is merged for 8.0 (and 7.13 as well), but the overall snapshot build is failing for master line since March 11 (when this was merged) so it makes sense why it is not picked up there yet.  last passing build:
https://internal-ci.elastic.co/view/Unified%20Release%20-%20snapshot/job/elastic+unified-release+master+distributed-snapshot-multijob-master/307/

closing as such.  please track in the above issue's PR's and if / when needed, watch the unified build process (like above) to see if the build has passed.  let me know any questions, thanks!

**Comment by dikshachauhan-qasource** [comment]
Hi Eric

Thank you for the feedback.

We will keep a track of [#24477](https://github.com/elastic/beats/issues/24477) and will validate whenever new build will be available.

Further, thanks for sharing follow up link for builds.
